datalad/local/foreach.py

```diff
@@ -0,0 +1,275 @@
+        #  on present subdatasets?  but having it explicit could be good to
+        #  "not miss any".  But `--fulfilled true` is getting on my nerves
+        fulfilled=Parameter(
+            args=("--fulfilled",),
+            doc="""if given, must be a boolean flag indicating whether
+            to report either only locally present or absent datasets.
+            By default subdatasets are reported regardless of their
+            status""",
+            constraints=EnsureBool() | EnsureNone()),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+        # does without any outputs handling
+        # Not sure yet if worth supporting here
+        # contains=Parameter(
+        #     args=('--contains',),
+        #     metavar='PATH',
+        #     action='append',
+        #     doc="""limit report to the subdatasets containing the
+        #     given path. If a root path of a subdataset is given the last
+        #     reported dataset will be the subdataset itself.[CMD:  This
+        #     option can be given multiple times CMD][PY:  Can be a list with
+        #     multiple paths PY], in which case datasets will be reported that
+        #     contain any of the given paths.""",
+        #     constraints=EnsureStr() | EnsureNone()),
+        # TODO: --diff  to provide `diff` record so any arbitrary  git reset --hard etc desire could be fulfilled
+        bottomup=Parameter(
+            args=("--bottomup",),
+            action="store_true",
+            doc="""whether to report subdatasets in bottom-up order along
+            each branch in the dataset tree, and not top-down."""),
+        # Extra options
+        # TODO: should we just introduce --lower-recursion-limit aka --mindepth of find?
+        subdatasets_only=Parameter(
+            args=("-s", "--subdatasets-only"),
+            action="store_true",
+            doc="""whether to exclude top level dataset."""),
+        passthrough=Parameter(  # TODO  could be of use for `run` as well
+            args=("-p", "--passthrough"),
+            action="store_true",
+            doc="""For command line commands, pass-through their output to 
+            the screen instead of capturing/returning as part of the result records."""),
+        jobs=jobs_opt,
+        # TODO: might want explicit option to either worry about 'safe_to_consume' setting for parallel
+        # For now - always safe
+    )
+
+    @staticmethod
+    @datasetmethod(name='foreach')
+    @eval_results
+    def __call__(
+            cmd=None,
+            python=False,
+            dataset=None,
+            fulfilled=None,
+            recursive=False,
+            recursion_limit=None,
+            # contains=None,
+            bottomup=False,
+            subdatasets_only=False,
+            passthrough=False,
+            jobs=None
+            ):
+        if not cmd:
+            lgr.warning("No command given")
```

Comment:
I think we have `InsufficientArgumentsError` or sth for that.

Reply:
done